src/number_theory/bernoulli.lean

```diff
@@ -390,4 +390,51 @@ begin
   field_simp [mul_right_comm _ ↑p!, ←mul_assoc _ _ ↑p!, cast_add_one_ne_zero, hne],
 end
 
+/-- Alternative form of Faulhaber's theorem, relating the sum of p-th powers with Bernoulli numbers.
+Deduced from `sum_range_pow`. -/
+theorem sum_range_pow' (n p : ℕ) :
+  ∑ k in Ico 1 (n + 1), (k : ℚ) ^ p
+    = ∑ i in range (p + 1), bernoulli' i * (p + 1).choose i * n ^ (p + 1 - i) / (p + 1) :=
+begin
+  -- We extract the first two terms of the sum on the right hand side and then replace instances of
+  -- `bernoulli'` by `bernoulli`, but the sum back together and apply `sum_range_pow`.
+  rw [sum_range_succ'],
+  -- if p is zero, there is no second term, so we make a special case
+  cases p with p hp,
+  { simp, },
+  -- extract the second term of the sum on the right hand side
+  { rw [sum_range_succ', ← bernoulli_eq_bernoulli'_of_ne_one zero_ne_one],
```

Comment:
When there's just one remaining case which takes a lot more work to prove than the earlier cases, we don't always put it inside braces. (If we do use braces then we usually add a layer of indentation.)
```suggestion
  rw [sum_range_succ', ← bernoulli_eq_bernoulli'_of_ne_one zero_ne_one],
```